trace-utils/src/trace_utils.rs

```diff
@@ -40,6 +45,210 @@ pub async fn get_traces_from_request_body(body: Body) -> anyhow::Result<(usize,
     Ok((size, traces))
 }
 
+#[inline]
+fn get_v5_strings_dict(reader: &mut Reader<impl Buf>) -> anyhow::Result<Vec<String>> {
+    let dict_size =
+        read_array_len(reader).map_err(|err| anyhow!("Error reading dict size: {err}"))?;
+    if dict_size as i32 > 25 * 1e6 as i32 {
+        anyhow::bail!(
+            "Error deserializing strings dictionary. Dict size is too large: {dict_size}"
+        );
+    }
+    let mut dict: Vec<String> = Vec::with_capacity(dict_size.try_into()?);
+    for _ in 0..dict_size {
+        let val: Value = read_value(reader)?;
+        match val {
+            Value::String(s) => {
+                let s = s.to_string().replace(&['\"'][..], "");
+                dict.push(s);
+            }
+            _ => anyhow::bail!("Error deserializing strings dictionary. Value in string dict is not a string: {val}")
+        }
+    }
+    Ok(dict)
+}
+
+#[inline]
+fn get_v5_span(reader: &mut Reader<impl Buf>, dict: &[String]) -> anyhow::Result<Span> {
+    let mut span: Span = Default::default();
+    let span_size = rmp::decode::read_array_len(reader)
+        .map_err(|err| anyhow!("Error reading span size: {err}"))? as usize;
+    if span_size != 12 {
+        anyhow::bail!("Expected an array of exactly 12 elements in a span, got {span_size}");
+    }
+    //0 - service
+    match read_value(reader)? {
+        Value::Integer(s) => {
+            span.service = str_from_dict(dict, s)?;
+        },
+        val => anyhow::bail!("Error reading span service, value is not an integer and can't be looked up in dict: {val}")
+    };
```

Comment:
This repeated code could be moved out to
```rust
fn get_v05_string(reader: &mut Reader<impl Buf>, dict: &[String], field_name: &str) -> anyhow::Result<String> {
    match read_value(reader)? {
        Value::Integer(s) => {
            str_from_dict(dict, s)
        },
        val => anyhow::bail!("Error reading {field_name}, value is not an integer and can't be looked up in dict: {val}")
    }
}
```

Reply:
Done